generator/transit_generator.cpp

```diff
@@ -275,50 +275,89 @@ void GraphData::DeserializeFromJson(my::Json const & root, OsmIdToFeatureIdsMap
   Visit(deserializer);
 }
 
-void GraphData::SerializeToMwm(string const & mwmPath) const
+void GraphData::SerializeToMwm(Writer & writer) const
 {
-  FilesContainerW cont(mwmPath, FileWriter::OP_WRITE_EXISTING);
-  FileWriter w = cont.GetWriter(TRANSIT_FILE_TAG);
-
   TransitHeader header;
 
-  auto const startOffset = w.Pos();
-  Serializer<FileWriter> serializer(w);
-  FixedSizeSerializer<FileWriter> numberSerializer(w);
+  auto const startOffset = writer.Pos();
+  Serializer<Writer> serializer(writer);
+  FixedSizeSerializer<Writer> numberSerializer(writer);
   numberSerializer(header);
-  header.m_stopsOffset = base::checked_cast<uint32_t>(w.Pos() - startOffset);
+  header.m_stopsOffset = base::checked_cast<uint32_t>(writer.Pos() - startOffset);
 
   serializer(m_stops);
-  header.m_gatesOffset = base::checked_cast<uint32_t>(w.Pos() - startOffset);
+  header.m_gatesOffset = base::checked_cast<uint32_t>(writer.Pos() - startOffset);
 
   serializer(m_gates);
-  header.m_edgesOffset = base::checked_cast<uint32_t>(w.Pos() - startOffset);
+  header.m_edgesOffset = base::checked_cast<uint32_t>(writer.Pos() - startOffset);
 
   serializer(m_edges);
-  header.m_transfersOffset = base::checked_cast<uint32_t>(w.Pos() - startOffset);
+  header.m_transfersOffset = base::checked_cast<uint32_t>(writer.Pos() - startOffset);
 
   serializer(m_transfers);
-  header.m_linesOffset = base::checked_cast<uint32_t>(w.Pos() - startOffset);
+  header.m_linesOffset = base::checked_cast<uint32_t>(writer.Pos() - startOffset);
 
   serializer(m_lines);
-  header.m_shapesOffset = base::checked_cast<uint32_t>(w.Pos() - startOffset);
+  header.m_shapesOffset = base::checked_cast<uint32_t>(writer.Pos() - startOffset);
 
   serializer(m_shapes);
-  header.m_networksOffset = base::checked_cast<uint32_t>(w.Pos() - startOffset);
+  header.m_networksOffset = base::checked_cast<uint32_t>(writer.Pos() - startOffset);
 
   serializer(m_networks);
-  header.m_endOffset = base::checked_cast<uint32_t>(w.Pos() - startOffset);
+  header.m_endOffset = base::checked_cast<uint32_t>(writer.Pos() - startOffset);
 
   // Rewriting header info.
   CHECK(header.IsValid(), (header));
-  auto const endOffset = w.Pos();
-  w.Seek(startOffset);
+  auto const endOffset = writer.Pos();
+  writer.Seek(startOffset);
   numberSerializer(header);
-  w.Seek(endOffset);
+  writer.Seek(endOffset);
 
   LOG(LINFO, (TRANSIT_FILE_TAG, "section is ready. Header:", header));
 }
 
+void GraphData::DeserializeForTesting(MemReader & reader)
```

Comment:
Why not to Deserialize from Reader?

Reply:
But if I do so I need to use the abstract Reader for constructing source: `ReaderSource<SomeReader>`. Which type shell I write instead of SomeReader in this case? In short I didn't manage to used Reader here.

Reply:
Done